### PR TITLE
Make Angular2 transformers lazy in debug mode when`lazy_transformers: true`

### PR DESCRIPTION
**Pull request by ochafik**
This make apps to load quicker with pub serve (only builds what is needed).

Note that lazy transformers seem to make pub build slower, so we wrap
transformers to force them to be eager in release mode.

### CODE REVIEW

File: modules_dart/transform/lib/src/transform/common/options.dart [hunk 1]
```diff
@@ -56,6 +57,13 @@ class TransformerOptions {
   /// at any time.
   final bool inlineViews;
 
+  /// Whether to make transformers lazy. views.
```
Comment (kegluneq) [inline modules_dart/transform/lib/src/transform/common/options.dart]: Remove trailing " views."

File: modules_dart/transform/lib/src/transform/stylesheet_compiler/transformer.dart [hunk 2]
```diff
@@ -21,12 +21,30 @@ class StylesheetCompiler extends Transformer {
   }
 
   @override
+  declareOutputs(DeclaringTransform transform) {
+    // Note: we check this assumption below.
+    _getExpectedOutputs(transform.primaryId).forEach(transform.declareOutput);
+  }
+
+  List<AssetId> _getExpectedOutputs(AssetId cssId) => [
+      cssId.addExtension('.dart'),
```
Comment (kegluneq) [inline modules_dart/transform/lib/src/transform/stylesheet_compiler/transformer.dart]: I'd prefer to reuse `to*StylesheetExtension`, but I don't feel too strongly about it.

https://github.com/angular/angular/blob/master/modules_dart/transform/lib/src/transform/common/names.dart#L61

File: modules_dart/transform/lib/src/transform/transformer.dart [hunk 3]
```diff
@@ -42,6 +43,11 @@ class AngularTransformerGroup extends TransformerGroup {
         ],
       ];
     }
+    if (options.modeName == BarbackMode.RELEASE || !options.lazyTransformers) {
+      phases = phases
+          .map((phase) => phase.map((t) => new EagerTransformerWrapper(t)))
+          .toList();
```
Comment (kegluneq) [inline modules_dart/transform/lib/src/transform/transformer.dart]: nit: `phases` is an `Iterable`, so no need for `toList()`

**Comment by kegluneq** [comment]
This looks great! Just a few minor comments.

**Comment by ochafik** [comment]
Thanks!
Adressed comments, rebasing soon :-)

**Comment by ochafik** [comment]
Ran dartfmt + squashed into 1 commit.

**Pull request closed without merging**